### PR TITLE
Provide alternative to Chef server search for splunk servers

### PR DESCRIPTION
**Pull request by Sauraus**
Fixes: https://github.com/chef-cookbooks/chef-splunk/issues/27

### CODE REVIEW

**Comment by danieljimenez** [comment]
Is this only used for downloading the spunk forwarder? If so, it worked great in my testing on my staging servers!

File: recipes/client.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@
   |a, b| a.name <=> b.name
 end
 
+if splunk_servers.empty?
+  splunk_servers = node['splunk']['splunk_servers']
+end
```
Comment (jtimberman) [inline recipes/client.rb]: Let's do an inline conditional.

And have a spec test that validates that this works.

**Comment by vidkun** [comment]
Any chances of this getting merged as is?

**Comment by tas50** [comment]
We'd like to see a spec to mock out an empty search and attribute usage before we merge this in.

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: recipes/client.rb (hunk 1 above)
Comment (tschroeder-zendesk) [inline recipes/client.rb]: You can make this one line as follows.
```
splunk_servers = node['splunk']['splunk_servers'] if splunks_servers.empty?
```

File: recipes/client.rb [hunk 2]
```diff
@@ -31,6 +31,14 @@
   a.name <=> b.name
 end
 
+if splunk_servers.empty?
+  splunk_servers = node['splunk']['splunk_servers']
+end
+
+if splunk_servers.empty?
+  splunk_servers = node['splunk']['splunk_servers']
+end
```
Comment (tschroeder-zendesk) [inline recipes/client.rb]: This is a duplicate block

**Comment by tschroeder-zendesk** [comment]
Can the duplicate block be fixed and the if changed to a one liner so this can get merged? I really need this feature ASAP.

**Comment by rarsan** [comment]
This PR seems to have been abandoned. Closing.
Please refer to new PR #93 with fix and unit/integration tests.

**Pull request closed without merging**